backend/bin/deal-observer-backend.js

```diff
@@ -40,21 +54,59 @@ const dealObserverLoop = async (makeRpcRequest, pgPool) => {
       Sentry.captureException(e)
     }
     const dt = Date.now() - start
-    console.log(`Loop "${LOOP_NAME}" took ${dt}ms`)
+    console.log(`Loop "${DEAL_OBSERVER_LOOP_NAME}" took ${dt}ms`)
+
+    if (INFLUXDB_TOKEN) {
+      recordTelemetry(`loop_${slug(DEAL_OBSERVER_LOOP_NAME, '_')}`, point => {
+        point.intField('interval_ms', DEAL_OBSERVER_LOOP_INTERVAL)
+        point.intField('duration_ms', dt)
+      })
+    }
+    if (dt < DEAL_OBSERVER_LOOP_INTERVAL) {
+      await timers.setTimeout(DEAL_OBSERVER_LOOP_INTERVAL - dt)
+    }
+  }
+}
+
+/**
+ * Periodically fetches unsubmitted deals from the database and submits them to the Spark API.
+ *
+ * @param {PgPool} pgPool
+ * @param {object} args
+ * @param {string} args.sparkApiBaseUrl
+ * @param {string} args.dealIngestionAccessToken
+ * @param {number} args.dealSubmitterBatchSize
+ */
+const sparkApiDealSubmitterLoop = async (pgPool, { sparkApiBaseUrl, dealIngestionAccessToken, dealSubmitterBatchSize }) => {
+  const submitDeals = submitDealsToSparkApi(sparkApiBaseUrl, dealIngestionAccessToken)
```

Comment:
I like the idea of having a `submitDeals` callback that accepts only `unsubmittedDeals` 👍🏻 

However, I expect a function called `submitDealsToSparkApi` to submit the deals, not to create another function.

Can we improve the name to make this clear, please? For example, we can prefix the function name with `create` or add a `Factory` suffix to the name.

For example:

```suggestion
  const submitDeals = createFnSubmitDealsToSparkApi(sparkApiBaseUrl, dealIngestionAccessToken)
```

Few more alternatives:
- createSubmitDealsToSparkApiCallback
- createSubmitDealsToSparkApiCb

If we cannot find a good name, then I think it's a sign we are creating a wrong abstraction.

An alternative solution to consider: curry the function here.

```js
const submitDeals = submitDealsToSparkApi(unsubmittedDeals, sparkApiBaseUrl, dealIngestionAccessToken)
```

Or even this:

```js
await findAndSubmitUnsubmittedDeals(
  pgPool, 
  dealSubmitterBatchSize,
  (unsubmittedDeals) => submitDealsToSparkApi(unsubmittedDeals, sparkApiBaseUrl, dealIngestionAccessToken)
)

Reply:
I have removed the currying